### ISSUE TITLE
m_chanhistory: Use message tags

### ISSUE THREAD

**Issue by docwhat** [inspircd/inspircd#1029]
In order to allow scripts, clients, etc. to handle history replay's correctly, the `m_chanhistory` module should use [message tags](https://github.com/ircv3/ircv3-specifications/blob/master/specification/message-tags-3.2.md) to properly identify the replayed history.

To see an example of how it can impact client developers, see [cinch issue 193](https://github.com/cinchrb/cinch/issues/193#issuecomment-86828512).

It would also be nice if [server time](https://github.com/ircv3/ircv3-specifications/blob/master/extensions/server-time-3.2.md) was implemented as well.

Note: I'm not super familiar with the IRC protocol and am relaying a feature request described by @dominikh. He may be able to provide more information if technical details are needed.

**Comment by SaberUK** [comment]
I don't think this will be possible for 2.2 as we need to rework all of our internal message handling stuff for message tags. Maybe 2.3.

**Comment by docwhat** [comment]
Sounds great!

**Comment by attilamolnar** [comment]
This is trivial to implement because of the way how m_chanhistory works, it sends raw messages that we can format differently for server-time capable clients.

**Comment by Heufneutje** [comment]
:+1:

**Comment by kaniini** [comment]
Based on the initial comment on this issue, I interpret it to mean he wants both server-time tag and another tag to describe that it's specifically history.

Other than `server-time`, what kind of tag are you thinking of?

**Comment by SaberUK** [comment]
@kaniini Presumably the tag/commands from `batch-3.2`.

**Comment by kaniini** [comment]
Oh, he didn't mention `BATCH`.  Okay, that makes sense.

**Comment by SadieCat** [comment]
This was resolved in v3 but I missed the issue, oops!
